org.adempiere.base/src/org/compiere/model/MOrder.java

```diff
@@ -2217,7 +2217,7 @@ && getC_Payment_ID() == 0 && getC_CashLine_ID() == 0)
 		//	Create SO Invoice - Always invoice complete Order
 		if ( MDocType.DOCSUBTYPESO_POSOrder.equals(DocSubTypeSO)
 			|| MDocType.DOCSUBTYPESO_OnCreditOrder.equals(DocSubTypeSO) 	
-			|| MDocType.DOCSUBTYPESO_PrepayOrder.equals(DocSubTypeSO)) 
+			|| MDocType.DOCSUBTYPESO_PrepayOrder.equals(DocSubTypeSO) && dt.isAutoGenerateInvoice()) 
```

Comment:
and also here - the call below to createPOSPayments() - line 2231 - will probably fail if the invoice is not autogenerated.

Reply:
With the fixed () in the last condition, createPOSPayments should return null for PrepayOrder which should be fine. For POSOrder the Invoice should be autogenerated.